src/lib/isEmail.js

```diff
@@ -24,10 +25,16 @@ export default function isEmail(str, options) {
   assertString(str);
   options = merge(options, default_email_options);
 
+  if (options.require_display_name) {
+    options.allow_display_name = true;
+  }
+
   if (options.allow_display_name) {
```

Comment:
I'd use `if (options.allow_display_name || options.require_display_name)` here instead of setting the flag above.